karma.conf.js

```diff
@@ -21,7 +21,23 @@ module.exports = function(config) {
                 loaders: [
                     {
                         test: /\.ts?$/,
-                        loader: 'ts-loader',
+                        loaders: [{
+                            loader: 'babel-loader',
+                            options: {
+                                presets: [
+                                    ['latest', {es2015: {modules: false}}],
+                                ],
+                            },
+                        }, 'ts-loader'],
+                    },
+                    {
+                        test: /\.js?$/,
+                        loader: 'babel-loader',
+                        options: {
+                            presets: [
+                                [ 'latest', { es2015: { modules: false } } ],
+                            ],
```

Comment:
why did you have to use babel-loader twice both for .ts and .js file?

Reply:
oh! this one bit me before!
Because if am importing both TS and JS files through the entry point, so the TS loader skips those files because they don't match the regex, doing this makes sure that babel processes JS files. Without this second rule, files are literally just concatenated, which surprisingly is not a problem in Edge, Chrome, Firefox and Safari as all of them support ES6 natively, but IE11... well... you know.